src/pages/library/RequestDatasetAccessModal.js

```diff
@@ -12,71 +12,12 @@ import * as Utils from 'src/libs/utils'
 
 
 const sendCopyEnabled = false
+const requestAccessEnabled = false
 
 export const RequestDatasetAccessModal = ({ onDismiss, datasets }) => {
   const [reason, setReason] = useState('')
   const [sendCopy, setSendCopy] = useState(false)
-
-  return h(Modal, {
-    title: 'Request Access',
-    width: '40rem',
-    showButtons: false,
-    showX: true,
-    onDismiss
-  }, [
-    div([
-      'You cannot access this dataset because it is protected by an Authorization Domain. ',
-      'You need to obtain permission from the owner(s) of this dataset in order to get access. ',
-      'Clicking the "Request Access" button below will send an email to the admins of that dataset.'
-    ]),
-    h(IdContainer, [id => div([
-      h(FormLabel, { htmlFor: id }, ['Please tell us why']),
-      h(TextArea, {
-        id,
-        style: { height: 100, marginBottom: '0.5rem' },
-        placeholder: 'Enter your reason',
-        value: reason,
-        onChange: setReason
-      }),
-      sendCopyEnabled && div([
-        h(LabeledCheckbox, {
-          checked: sendCopy,
-          onChange: setSendCopy
-        }, [
-          label({ style: { margin: '0 2rem 0 0.25rem' } }, [`Also send me a copy`])
-        ])
-      ])
-    ])]),
-    table({ style: { margin: '1rem', width: '100%' } }, [
-      thead([
-        tr({ style: { height: '2rem' } }, [th({ style: { textAlign: 'left' } }, ['Datasets']), th({ style: { textAlign: 'left', width: '15rem' } }, ['Access'])])
-      ]),
-      tbody(
-        _.map(({ 'dct:title': title, access, 'dct:identifier': id, contacts }) => tr({ key: id, style: { height: '2rem' } }, [
-          td({ style: { paddingRight: 20 } }, [
-            title,
-            div({ style: { fontSize: '.7rem', marginTop: 5, width: 'fit-content' } }, [
-              _.map(({ email }) => email && h(Link, { key: email, href: `mailto:${email}`, style: { marginTop: 5, display: 'block' } }, [email]), contacts)
-            ])
-          ]),
-          td([
-            Utils.switchCase(access,
-              ['Controlled', () => h(RequestDatasetAccessButton, { title, id })],
-              ['Pending', () => span({ style: { fontWeight: 600 } }, ['Request Pending'])],
-              [Utils.DEFAULT, () => span({ style: { fontWeight: 600 } }, ['Permission Granted'])]
-            )
-          ])
-        ]), datasets)
-      )
-    ])
-  ])
-}
-
-const RequestDatasetAccessButton = ({ title, id }) => {
-  const [requesting, setRequesting] = useState(false)
-  const [requested, setRequested] = useState(false)
   const [showWipModal, setShowWipModal] = useState(false)
-  const signal = Utils.useCancellation()
 
   return h(Fragment, [
```

Comment:
The traditional method to do this would be, instead of having both of these modals in a fragment, to either write it as a ternary. I also wouldn't expect to have to call `setShowWipModal(false)` as well as `onDismiss`, since the latter should unmount the entire component.

Reply:
good point, thank you!